src/main/java/com/iexec/commons/poco/chain/Web3jAbstractService.java

```diff
@@ -261,36 +274,7 @@ public BigInteger getUserGasPrice(float gasPriceMultiplier, long gasPriceCap) {
         return BigInteger.valueOf(Math.min(wishedGasPrice, gasPriceCap));
     }
 
-    /*
-     * This is just a dummy stub for contract reader:
-     * gas price & gas limit is not required when querying (read) an eth node
-     *
-     */
-    public ContractGasProvider getReadingContractGasProvider() {
-        return new ContractGasProvider() {
-            @Override
-            public BigInteger getGasPrice(String contractFunc) {
-                return null;
-            }
-
-            @Override
-            public BigInteger getGasPrice() {
-                return null;
-            }
-
-            @Override
-            public BigInteger getGasLimit(String contractFunc) {
-                return null;
-            }
-
-            @Override
-            public BigInteger getGasLimit() {
-                return null;
-            }
-        };
-    }
-
-    public ContractGasProvider getWritingContractGasProvider() {
+    private ContractGasProvider getWritingContractGasProvider() {
```

Comment:
We may create a bean out of this `ContractGasProvider` in a next PR. It would make `Web3jAbstractService` easier to test.